maven35-agent/pom.xml

```diff
@@ -24,12 +24,48 @@
     <dependency>
       <groupId>org.codehaus.plexus</groupId>
       <artifactId>plexus-classworlds</artifactId>
+      <version>2.5.2</version>
     </dependency>
     <dependency>
       <groupId>org.eclipse.sisu</groupId>
       <artifactId>org.eclipse.sisu.plexus</artifactId>
-      <version>0.0.0.M2a</version>
+      <version>0.3.3</version>
     </dependency>
+    
+   <!-- Upper bounds from org.apache.maven:maven-embedder:3.5.0 -->
```

Comment:
depends on how we use this lib... The problem with Maven interceptors is that this library may be used outside Maven plugin, which just bundles all agents and interceptors as dependencies (what could possibly go wrong?). 

If we do not care about such usages, we could align all versions via parent POM. It needs weight in from @aheritier and @olamy